test/collections.js

```diff
@@ -145,6 +145,23 @@
     });
   });
 
+  test('Iterate function as object', function() {
+    //github issue #1992: collection methods should not iterate functions as arrays but as objects
+
+    var collection_methods = ['each','map','reduce','reduceRight','find','filter','where','findWhere','reject','every','some','contains','pluck','max','min','sortBy','groupBy','indexBy','countBy','shuffle','sample','toArray','size','partition'];
+    collection_methods.map(iterate_function_as_object);
```

Comment:
Hi, couple nits here:

- you can't use `methods.map` here as we test in ES3 environments (use `_.each` instead)
- camel case methods
- add a space between the comman and strings